tests/dex-crds-base_test.go

```diff
@@ -0,0 +1,322 @@
+package tests_test
+
+import (
+	"sigs.k8s.io/kustomize/k8sdeps/kunstruct"
+	"sigs.k8s.io/kustomize/k8sdeps/transformer"
+	"sigs.k8s.io/kustomize/pkg/fs"
+	"sigs.k8s.io/kustomize/pkg/loader"
+	"sigs.k8s.io/kustomize/pkg/resmap"
+	"sigs.k8s.io/kustomize/pkg/resource"
+	"sigs.k8s.io/kustomize/pkg/target"
+	"testing"
+)
+
+func writeDexCrdsBase(th *KustTestHarness) {
+	th.writeF("/manifests/common/dex-auth/dex-crds/base/namespace.yaml", `
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: auth
+`)
+	th.writeF("/manifests/common/dex-auth/dex-crds/base/config-map.yaml", `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dex
+data:
+  config.yaml: |
+    issuer: $(issuer)
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      https: 0.0.0.0:5556
+      tlsCert: /etc/dex/tls/tls.crt
```

Comment:
I believe the current practice is to do tls termination at the Istio Gateway / Ingress and not use certificates for each individual application.
Because of that, can we remove this setting or put it in an overlay?
If we need additional security in the future, we can use Istio for mTLS in the kubeflow namespace.

Reply:
I understand. I've suggested an alternative for the base to just contain a mock callback and the overlay to have LDAP. Though the challenge here is that we may need to provide tooling to overlay instead of adding them to manifests.